core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java

```diff
@@ -185,20 +189,24 @@ private void buildMdcContext(Stack stack) {
     }
 
     private void reportHealthAndSyncInstances(Stack stack, Collection<InstanceMetaData> runningInstances, Collection<InstanceMetaData> failedInstances,
-            Set<String> newHealtyHostNames, InstanceSyncState defaultState) {
+            Set<String> newHealtyHostNames, boolean hostCertExpiring) {
         Set<String> newFailedNodeNames = failedInstances.stream()
                 .filter(i -> !Set.of(SERVICES_UNHEALTHY, STOPPED).contains(i.getInstanceStatus()))
                 .map(InstanceMetaData::getDiscoveryFQDN)
                 .collect(toSet());
-        ifFlowNotRunning(() -> clusterOperationService.reportHealthChange(stack.getResourceCrn(), newFailedNodeNames, newHealtyHostNames));
-        ifFlowNotRunning(() -> {
-            if (!failedInstances.isEmpty()) {
-                clusterService.updateClusterStatusByStackId(stack.getId(), Status.AMBIGUOUS);
-            } else if (statesFromAvailabeAllowed().contains(stack.getCluster().getStatus())) {
-                clusterService.updateClusterStatusByStackId(stack.getId(), Status.AVAILABLE);
-            }
-        });
-        syncInstances(stack, runningInstances, failedInstances, defaultState);
+        ifFlowNotRunning(() -> updateStates(stack, failedInstances, newFailedNodeNames, newHealtyHostNames, hostCertExpiring));
+        syncInstances(stack, runningInstances, failedInstances, InstanceSyncState.RUNNING);
+    }
+
+    private void updateStates(Stack stack, Collection<InstanceMetaData> failedInstances, Set<String> newFailedNodeNames, Set<String> newHealtyHostNames,
+            boolean hostCertExpiring) {
+        clusterService.updateClusterCertExpirationState(stack.getCluster(), hostCertExpiring);
+        clusterOperationService.reportHealthChange(stack.getResourceCrn(), newFailedNodeNames, newHealtyHostNames);
+        if (!failedInstances.isEmpty()) {
```

Comment:
Can you please add a few log lines here?

Reply:
added